src/workspaceViewer.ts

```diff
@@ -0,0 +1,143 @@
+import path = require('path');
+import { TreeDataProvider, EventEmitter, TreeItemCollapsibleState, TreeItem, Event, Uri, window } from 'vscode';
+import { runTextInTerm } from './rTerminal';
+import { globalenv, workingDir } from './session';
+import { config } from './util';
+
+interface WorkspaceAttr {
+    [key: string]: {
+        class: string[];
+        type: string;
+        length: number;
+        str: string;
+    }
+}
+export class WorkspaceDataProvider implements TreeDataProvider<WorkspaceItem> {
+	private _onDidChangeTreeData: EventEmitter<void> = new EventEmitter();
+	readonly onDidChangeTreeData: Event<void> = this._onDidChangeTreeData.event;
+
+	refresh(): void {
+		this.data = <WorkspaceAttr> globalenv;
+		this._onDidChangeTreeData.fire();
+	}
+
+	data: WorkspaceAttr;
+
+	getTreeItem(element: WorkspaceItem): TreeItem {
+		return element;
+	}
+
+	getChildren(): WorkspaceItem[] {
+		return this.getWorkspaceItems(this.data);
+	}
+
+	private getWorkspaceItems(data: WorkspaceAttr): WorkspaceItem[] {
+		const toItem = (
+			key: string,
+			rClass: string,
+			str: string,
+			type: string,
+			length: number
+		): WorkspaceItem => {
+			return new WorkspaceItem(
+				key,
+				rClass,
+				str,
+				type,
+				length,
+				TreeItemCollapsibleState.None
+			);
+		};
+
+		const items = data ? Object.keys(data).map((key) =>
+			toItem(
+				key,
+				data[key].class[0],
+				data[key].str,
+				data[key].type,
+				data[key].length
+			)) : [];
+
+		function sortItems(a: WorkspaceItem, b: WorkspaceItem) {
+			const priorityAttr: string[] = [
+				'data.frame',
+				'data.table',
+				'environment',
+				'list',
+				'tibble',
+				'tbl_df',
+				'tbl'
+			]
+
+			return priorityAttr.includes(a.contextValue) > priorityAttr.includes(b.contextValue) ? -1 : priorityAttr.includes(b.contextValue) > priorityAttr.includes(a.contextValue) ? 1 : 0 || a.label.localeCompare(b.label);
```

Comment:
This line is a bit too long. Would you like to rewrite it in `if else` form instead?